src/pack_unpack.c

```diff
@@ -291,6 +290,26 @@ static json_t *pack_array(scanner_t *s, va_list *ap)
     return NULL;
 }
 
+static json_t *pack_string(scanner_t *s, va_list *ap) {
+    char *str;
+    size_t len;
+    int ours;
+    int nullable;
+
+    nullable = s->start[1] == '?';
+    if (nullable)
+        next_token(s);
```

Comment:
You should use `next_token()` and `prev_token()` instead of `s->start[1]`:

```c
next_token(s);
nullable = token(s) == '?';
if (!nullable) prev_token(s);
```